attendance/views.py

```diff
@@ -119,4 +131,62 @@ def get_item(dictionary, key):
 
 
 def get_date_from_template_returned_string(string_date):
-    return datetime.strptime(string_date, "%B %d, %Y").date()
+    for date_format in ["%B %d, %Y", "%b. %d, %Y", "%bt. %d, %Y"]:
+        try:
+            return datetime.strptime(string_date, date_format).date()
+        except ValueError:
+            pass
+    raise ValueError("time data {!r} does not match any expected date format".format(string_date))
+
+
+def update_course_attendance_statistic(course_id):
+    class_attendance = Attendance.objects.filter(
+        classroom_id=course_id, date__range=["2000-01-01", datetime.today().date()]
+    )
+    average = get_average_attendance_from_list(class_attendance)
+    classroom = Classroom.objects.get(id=course_id)
+    classroom.attendance_summary = {"attendance_statistic": round(average * 100, 2)}
+    classroom.save()
+
+
+def get_course_attendance_statistic(course_id):
+    return Classroom.objects.get(id=course_id).attendance_summary.get(
+        "attendance_statistic"
+    )
+
+
+def create_absence_notifications(request, student_list, absences, date):
+    notifications = [Notification(
+        subject="%s %s absence on %s" % (student.first_name, student.last_name, date),
+        notification=get_generic_absence_notification(student, date),
+        user_id=student.id,
+        attendance_id=absences[x].id,
+        created_by=DjangoUser.objects.get(id=request.user.id),
+        email_recipients=True
+    ) for x, student in enumerate(student_list)]
```

Comment:
If you were to need to do something like this in the future, [zip](https://docs.python.org/3/library/functions.html#zip) is better than using `enumerate` and indexing the second list.

It would make sense to lift `DjangoUser.objects.get(id=request.user.id)` out of the loop so it's not queried N times.

Reply:
Ahh yes I should've noticed that!

Reply:
Resolved in 989e97ca4de16df31e508a8eeb5154ade44b3bd5